src/processors/default/processTrackArtworks.ts

```diff
@@ -0,0 +1,115 @@
+    return new Promise(function (resolve, reject) {
+      im.convert([originalPath, '-resize', size, outPath],
+        async function (err, stdout) {
+          if (err) reject(err);
+          resizedBuffer = await readFile(outPath)
+          resolve(resizedBuffer);
+        });
+    })
+  }, 'png');
+  return buffer;
+}
+
+async function uploadBuffer(buffer: Buffer) {
+  const url: any = '/ip4/127.0.0.1/tcp/5011'; // have to use any because create only accepts hash
+  const client = create(url)
+  const { cid } = await client.add(buffer)
+  return cid;
+}
+
+async function withTempPath(callback: (path: string) => Promise<any>, ext: string | void): Promise<any> {
+  const fileName = (Math.random().toString(16) + '0000000').substr(2, 8);
+  const path = `${Path.join(tempDirectory, fileName)}${ext ? '.' + ext : ''}`;
+
+  try {
+    return await callback(path);
+  } catch (e) {
+    throw e;
+  } finally {
+    await rm(path);
+  }
+}
+
+async function temporaryWrite(buffer: Buffer, callback: (path: string) => Promise<void | Buffer>): Promise<void> {
+  return withTempPath(async function (path) {
+    await writeFile(path, buffer)
+    await callback(path);
+  })
+}
+// when i use ProcessedTrack it claims lossyArtworkURL does not exist :shrug:
+const processArtwork = async function (clients: Clients, nft: any): Promise<void> {
+  console.log(`processing ${nft.lossyArtworkURL}`)
+  const buffer = await imageBuffer(nft.lossyArtworkURL);
+  return temporaryWrite(buffer, async function (originalPath) {
+    console.log(`wrote image to ${originalPath}`)
+    let imageProcessingResults: any = null;
+    try {
+      const [largeImageBuffer, thumbnailImageBuffer]
+        = await Promise.all([resizeImage(originalPath, '700x700'), resizeImage(originalPath, '200x200')]);
+      const [coverCid, thumnailCid] = await Promise.all([uploadBuffer(largeImageBuffer), uploadBuffer(thumbnailImageBuffer)])
+      imageProcessingResults = [
+        { cid: coverCid.toString(), size: 'cover', trackId: nft.id },
+        { cid: thumnailCid.toString(), size: 'thumbnail', trackId: nft.id }
+      ]
+    } catch (e: any) {
+      imageProcessingResults = [
+        { error: e.toString(), size: 'cover', trackId: nft.id },
+        { error: e.toString(), size: 'thumbnail', trackId: nft.id }
+      ]
+    } finally {
+      await clients.db.insert(Table.processedArtworks, imageProcessingResults);
+    }
+  });
+}
+
+const processorFunction = async (items: Array<ProcessedTrack>, clients: Clients) => {
+  const boundPromiseCreator = processArtwork.bind(null, clients);
+  const maxPPM = parseInt(process.env.MAX_IMAGEMAGIC_PER_MINUTE!);
+  const maxCP = parseInt(process.env.MAX_CONCURRENT_IMAGEMAGIC!);
+  const results = await rollPromises<ProcessedTrack, void, Error>(items, boundPromiseCreator, maxCP, maxPPM);
+}
+
+export const processTrackArtworks = {
+  name: 'processTrackArtworks',
+  trigger: missingProcessedArtworks,
+  processorFunction: processorFunction,
+};
```

Comment:
do we care about new lines at end of file, i can set a rule for it

Reply:
whatevr

Reply:
lol, i am a fan as well. what editor are you using btw?

Reply:
vscode